src/compute-client/src/metrics.rs

```diff
@@ -113,6 +120,17 @@ impl ComputeControllerMetrics {
                 help: "The number of dataflows in the controller's command history.",
                 var_labels: ["instance_id"],
             )),
+            peeks_total: metrics_registry.register(metric!(
+                name: "mz_compute_peeks_total",
+                help: "The total number of peeks served.",
+                var_labels: ["instance_id", "result"],
+            )),
+            peek_duration_seconds: metrics_registry.register(metric!(
+                name: "mz_compute_peek_duration_seconds",
+                help: "A histogram of peek durations since restart.",
+                var_labels: ["instance_id", "result"],
+                buckets: histogram_seconds_buckets(0.000_500, 32.),
```

Comment:
For the lower bound I looked at what existing code is already using. It seems unrealistic that peeks that involve network communication would be complete in less than one ms, so this should be fine.